src/types/form_logic.ts

```diff
@@ -51,3 +52,55 @@ export interface IPreventSubmitLogic extends ILogic {
 export interface IPreventSubmitLogicSchema
   extends IPreventSubmitLogic,
     Document {}
+
+type LogicField = Extract<
+  BasicField,
+  | BasicField.Dropdown
+  | BasicField.Radio
+  | BasicField.YesNo
+  | BasicField.Number
+  | BasicField.Decimal
+  | BasicField.Rating
+>
+
+type LogicCondition<K extends LogicField, V extends LogicConditionState> = [
+  K,
+  Array<V>,
```

Comment:
nit - we use `[]` notation rather than `Array<>` everywhere else in the repo
```suggestion
 V[],
```